### DISCUSSION TITLE
New install in eclipse, palette won't work

### DISCUSSION THREAD

**Discussion by peterkelley519** [eclipse/windowbuilder#724] [Q&A] [answered]
Multiple symptoms - 
I just added the Current build using Eclipse MarketPlace, and in a new Java project I created a Swing Application window class. 

1. Palette displays components fine, but when I hover over anything in the palette I am unable to choose any component. When I hover over any component in the palette the hover shows "JTabbedPane"(with desc) regardless of where the cursor entered the palette frame, or whatever component I'm hovering over. 
2. Because of the  I created a Swing Application window class. I can change the Layout using the Structure dialog[components/properties] and immediately see an error for **NoSuchMethodError - 'java.lang.Iterable org.eclipse.wb.gef.core.EditPart.getEditPolicyIterable()'**
3.  I found that the EclMarketplace had installed both the Current and the "WindowBuilder Nightly Build Nightly"(sic) . I was sure that I had only asked to install the Current only. Uninstalled both and reinstalled selecting only the Current, and the same thing happened - both were installed even though I only asked for the Current version.  

I'm trying to debug this further and collect more details, but I must ask the Community if the #3 issue is somehow related to my overall problems? How can I install only the Current build?
As for problems #1 and #2 I'll seek more log info, but won't mind any helpful pointers. I've searched and found similar sounding issues in the past - but they differ in that the palette either won't display at all or displays empty, and get resolved by running an update(which I've done). 
I'm looking forward to learning more about this plugin.
Best regards!

Eclipse 2023-06,
JDK 11 - openjdk_ms_jdk-11.022+7

**Comment by ptziegler** [comment]
> NoSuchMethodError - 'java.lang.Iterable org.eclipse.wb.gef.core.EditPart.getEditPolicyIterable()'

It would be interesting to see which WindowBuilder bundles are actually installed in your IDE. This error seems to indicate that something wasn't properly installed. If there is something wrong with the installation, this could then describe both the first and the second problem.

In the menu bar, you can check under Help -> About Eclipse -> Features which features have been installed (and thus also the bundles). In a clean installation, those should all be in version 1.15.0.

>  "WindowBuilder Nightly Build Nightly"(sic) .

The second `Nightly` describes the version. I wonder if there is a better way to represent this...

> Eclipse 2023-06,
> JDK 11 - openjdk_ms_jdk-11.022+7

How are you able to use the Eclipse 2023-06 with Java 11, when it requires at least Java 17?
Does this problem also show up with the Eclipse 2024-03?

**Reply by peterkelley519** [reply]
Thank you for your assistance! I'm checking into more details - I really appreciate your insight, the direction is helpful. I'll be able to respond a bit later though with more info. For now, I can see that the installed versions of the WindowBuilder features are all at 1.16.202403132032, except for WindowBuilder SWT Designer which is at 1.16.202403132118.   
The above version 1.16.* instead of 1.15.*  might be related to the MarketPlace question I had - when I ask for the Current Version when it finishes I see that the Current and the Nightly Build are added into the installed MarketPlace section. Is that expected? 
Stuff I need to follow up with include - I'll check whether my install of Eclipse IDE 2023-06 was with JDK 17 ( I think it was ) but the particular project reqs are for Java 11 (I'll check to verify) - I need to dig a little more to see if I can find anything unusual in the logs. 
Thank you for the helpful info. The WindowBuilder project looks like such a useful tool and doubly so that it has active support! 
Cheers!

**Reply by peterkelley519** [reply]
I tried the uninstall/reinstall, and as before the installed section of MarketPlace shows both plugins. The version of all the components for WindowBuilder is at 1.15.* . I think that if I apply the "update" in marketplace that will bring me to 1.16.*, but I'll leave that for now. 
Other misc - >verified Ecl 2023-06 is running w/ JDK 17, and my project had been set to use a JDK 11. 
   - > I created a new project using JDK 17 , and the issue is still there.
  My thought is that this might get cleaned up for me by upgrading Eclipse to the current version. Unfortunately for my main work machine I have to hold off on that. As a temporary workaround I'll see if I can run WindowBuilder on a separate Eclipse install, and import the results of that into my main project. ( What I have in mind is rather straightforward, WB will save me a lot of effort regardless.)

**Reply by ptziegler** [reply]
> The above version 1.16.* instead of 1.15.* might be related to the MarketPlace question I had - when I ask for the Current Version when it finishes I see that the Current and the Nightly Build are added into the installed MarketPlace section. Is that expected?

1.16.* is the nightly build, while 1.15.* is the current release. Why it installed both when you only selected "Current", I can't say and it's also not something I am able to reproduce.

> verified Ecl 2023-06 is running w/ JDK 17, and my project had been set to use a JDK 11.

That's what I expected. The interesting part should be the JDK that is used by the Eclipse, because that's also what's used by the WindowBuilder.

If the problem still shows up, it might also be worth checking out the version of GEF that is installed (which declares the missing getEditPolicyIterable() method). Ideally, it should be 3.15.0.202402212051

If that is the case, I would also double-check whether not just the features, but also the plugins are are in the correct version.
![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/70652978/09797f02-57a1-42cd-8134-83080284e581)

If everything matches, then I would really everything again with a vanilla installation of the Eclipse 2024-03.

**Reply by peterkelley519** [reply]
Your recommendation to try it on a vanilla install seems the most sensible approach. Currently I cannot yet do that with my workspace for a while since other plugins are needed for my project(Gradle, Web, etc). When I get that project settled I will update stuff. 

Re Eclipse GEF - I checked the version via Eclipse features window - 3.16.0.202306071707 for the two features GEF Classic Draw2d and GEF Classic GEF (MVC) .  Odd thing is that when I click on the plug-in details button for those same plugins in the same Eclipse features window it shows versions lower than 3.15 . 

On a separate virtual machine (virtualbox) I'm currently using the plugin just fine, and I've imported the results back to my original machine and project, where the resulting classes work just as expected.  That's my temporary workaround. 
Best regards! (and thanks!)

**Comment by ptziegler** [comment, marked as answer]
> Re Eclipse GEF - I checked the version via Eclipse features window - 3.16.0.202306071707 for the two features GEF Classic Draw2d and GEF Classic GEF (MVC) . Odd thing is that when I click on the plug-in details button for those same plugins in the same Eclipse features window it shows versions lower than 3.15 .

And there's the problem. The method `getEditPolicyIterable()` was introduced with https://github.com/eclipse/gef-classic/commit/509370ee34f9579bc81cd6aaea6200ce83f46d8c, i.e. as part of the 2023-09 release. So the version that is installed in your IDE is not compatibly with the latest WindowBuilder version, where the lower bound is not correctly set, to reflect this.

The easiest way to fix this is by manually updating GEF and chances are high that WindowBuilder will then work correctly.
Click on "Help" in the menu bar, then go "Install New Software...". In the dialog, paste the following URL:
https://download.eclipse.org/tools/gef/classic/release/latest/

Then select both Draw2D and GEF. This should then lead to the installation of the plugins from the 2024-03 release, which is what WindowBuilder is tested against.

![image](https://github.com/eclipse-windowbuilder/windowbuilder/assets/70652978/7c0d4822-ea5b-4ce9-bf68-56a1ca76d9af)

**Comment by peterkelley519** [comment]
Well that did it! Thanks for the help and the detailed explanation of the fix! 
I've summarized your response and help on a stackoverflow question that I posted earlier. (and linking back to this discussion) This was to share the solution if other folks had the same problem but didn't happen to check github. If you happen to have a StackOverflow acct and care to add a response I would tag that as the accepted answer.
Best regards!
